src/profiler/cpuProfilerModel.ts

```diff
@@ -0,0 +1,374 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
```

Comment:
Apache 2.0 recommends keeping a notice of changes for original code. If I'm reading this correctly, this file is a mix of Lighthouse code mixed with ours? If that's the case, let's discuss what's the best way to do put that in licensing terms.

Reply:
Yes, essentially I just converted all the code written there into TS and added a couple of functions of my own. Let's continue this on Discord

Reply:
@saphal1998 cool! I think that's fine. In the future, when you source code from elsewhere, it's usually good to layer the commits so that they're easier to review:

- 1/n commit: Just the original file
- 2/n commit: Convert to TypeScript
- 3-n/n commits: Add functionality

That way the code is easier to review (right now I have no idea who's written what), and those commits can be used in the license header to clearly show what changes have been made.

No need to do this retroactively, I'll review 👌